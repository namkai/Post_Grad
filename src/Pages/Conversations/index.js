import React from 'react';
import { connect } from 'react-redux';
import { selectedConversation } from './Actions/index';
import ConversationList from './Components/ConversationList';
import Messages from './Components/Messages';

const Conversations = ({ conversations, messages }) => {
  console.log('i\'m the messages! in CONVO', messages);
  return (
    <div className="modal fade" id="msgModal" tabIndex="-1" role="dialog" aria-labelledby="msgModal" aria-hidden="true">
      <div className="modal-dialog">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title">Conversations</h5>
            <button type="button" className="close" data-dismiss="modal" aria-hidden="true">&times;</button>
          </div>

          <div className="modal-body p-0 js-modalBody">
            <div className="modal-body-scroller">
              <div className="media-list media-list-users list-group js-msgGroup">
                <ConversationList conversations={conversations} selectedConversation={selectedConversation} />
              </div>
              <Messages messages={messages} />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default connect(
	({ conversations, user, messages }) => ({ conversations, user, messages }),
	{ selectedConversation })(Conversations);
