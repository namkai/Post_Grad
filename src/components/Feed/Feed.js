import React from 'react';
import ProfileAbout from './FeedAbout';
import ProfileModal from './FeedModal';
import requireAuth from '../HOC/Require_Authentication';

const Profile = () => {
	return (
		<div>
		<div className="container pt-4">
			<div className="row">
				<ProfileAbout/>
			</div>
		</div>
		</div>
	);
};

export default requireAuth(Profile);
