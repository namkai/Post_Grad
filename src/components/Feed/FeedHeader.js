import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import headshot from '../../../style/assets/headshot-namkai.jpg';

const FeedHeader = ({ user }) => (
	<div className="card card-profile mb-4">
		<div className="card-header" style={ { backgroundImage: `url(${ user.backgroundPhoto })` } }/>
		<div className="card-block text-center">
			<Link to="/profile">
				<img
					className="card-profile-img"
					src={ headshot }
				/>
			</Link>

			<h6 className="card-title">
				<a className="text-inherit" href="profile/index.html">{user.name}</a>
			</h6>

			<p className="mb-4">{user.bio}</p>

			<ul className="card-menu">
				<li className="card-menu-item">
					<a href="#userModal" className="text-inherit" data-toggle="modal">
						Friends
						<h6 className="my-0">{user.friends}</h6>
					</a>
				</li>

				<li className="card-menu-item">
					<a href="#userModal" className="text-inherit" data-toggle="modal">
						Projects
						<h6 className="my-0">100</h6>
					</a>
				</li>
			</ul>
		</div>
	</div>
);

export default connect(({ user }) => ({ user }))(FeedHeader);
