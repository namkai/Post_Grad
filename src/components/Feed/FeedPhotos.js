import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Grid, Row, Col, Image } from 'react-bootstrap';
import Photo from './Photo';

class ProfilePhotos extends Component {
	componentDidMount() {

	}

	render() {
		const generatePhotos = this.props.user.photos.map(photo => (<Photo photo={ photo } />));
		return (
			<div className="card visible-md-block visible-lg-block">
				<div className="card-block">
					<h6 className="mb-3">Photos
						<small>Â· <a href="#">Edit</a></small>
					</h6>

					<div className="row">
						<div data-grid="images" data-target-height="150">
							{generatePhotos}
						</div>
					</div>
				</div>
			</div>
		);
	}
}

export default connect(({ user }) => ({ user }))(ProfilePhotos);
