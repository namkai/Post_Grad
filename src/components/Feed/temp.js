{/*<div className="card visible-md-block visible-lg-block mb-4">*/}
{/*<div className="card-block">*/}
{/*<h6 className="mb-3">About*/}
{/*<small>· <a href="#">Edit</a></small>*/}
{/*</h6>*/}
{/*<ul className="list-unstyled list-spaced">*/}
{/*<li><span className="text-muted icon icon-calendar mr-3"/>Went to <a href="#">Oh, Canada</a></li>*/}
{/*<li><span className="text-muted icon icon-users mr-3"/>Became friends with <a*/}
{/*href="#">Obama</a></li>*/}
{/*<li><span className="text-muted icon icon-github mr-3"/>Worked at <a href="#">Github</a></li>*/}
{/*<li><span className="text-muted icon icon-home mr-3"/>Lives in <a href="#">San Francisco,*/}
{/*CA</a></li>*/}
{/*<li><span className="text-muted icon icon-location-pin mr-3"/>From <a href="#">Seattle,*/}
{/*WA</a></li>*/}
{/*</ul>*/}
{/*</div>*/}
{/*</div>*/}

{/*<div className="card visible-md-block visible-lg-block">*/}
{/*<div className="card-block">*/}
{/*<h6 className="mb-3">Photos*/}
{/*<small>· <a href="#">Edit</a></small>*/}
{/*</h6>*/}
{/*<div data-grid="images" data-target-height="150">*/}
{/*<div>*/}
{/*<img data-width="640" data-height="640" data-action="zoom" src="assets/img/instagram_5.jpg"/>*/}
{/*</div>*/}

{/*<div>*/}
{/*<img data-width="640" data-height="640" data-action="zoom" src="assets/img/instagram_6.jpg"/>*/}
{/*</div>*/}

{/*<div>*/}
{/*<img data-width="640" data-height="640" data-action="zoom" src="assets/img/instagram_7.jpg"/>*/}
{/*</div>*/}

{/*<div>*/}
{/*<img data-width="640" data-height="640" data-action="zoom" src="assets/img/instagram_8.jpg"/>*/}
{/*</div>*/}

{/*<div>*/}
{/*<img data-width="640" data-height="640" data-action="zoom" src="assets/img/instagram_9.jpg"/>*/}
{/*</div>*/}

{/*<div>*/}
{/*<img data-width="640" data-height="640" data-action="zoom" src="assets/img/instagram_10.jpg"/>*/}
{/*</div>*/}
{/*</div>*/}
{/*</div>*/}
{/*</div>*/}
{/*</div>*/}

{/*<div className="col-lg-6">*/}

{/*<ul className="list-group media-list media-list-stream mb-4">*/}

{/*<li className="media list-group-item p-4">*/}
{/*<div className="input-group">*/}
{/*<input type="text" className="form-control" placeholder="Message"/>*/}
{/*<div className="input-group-btn">*/}
{/*<button type="button" className="btn btn-secondary">*/}
{/*<span className="icon icon-camera"/>*/}
{/*</button>*/}
{/*</div>*/}
{/*</div>*/}
{/*</li>*/}

{/*<li className="media list-group-item p-4">*/}
{/*<img*/}
{/*className="media-object d-flex align-self-start mr-3"*/}
{/*src="assets/img/avatar-dhg.png"/>*/}
{/*<div className="media-body">*/}
{/*<div className="media-heading">*/}
{/*<small className="float-right text-muted">4 min</small>*/}
{/*<h6>Dave Gamache</h6>*/}
{/*</div>*/}

{/*<p>*/}
{/*Aenean lacinia bibendum nulla sed consectetur. Vestibulum id ligula porta felis euismod semper.*/}
{/*Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Cras justo odio, dapibus ac*/}
{/*facilisis in, egestas eget quam. Vestibulum id ligula porta felis euismod semper. Cum sociis*/}
{/*natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.*/}
{/*</p>*/}

{/*<div className="media-body-inline-grid" data-grid="images">*/}
{/*<div style="display: none">*/}
{/*<img data-action="zoom" data-width="1050" data-height="700" src="assets/img/unsplash_1.jpg"/>*/}
{/*</div>*/}

{/*<div style="display: none">*/}
{/*<img data-action="zoom" data-width="640" data-height="640" src="assets/img/instagram_1.jpg"/>*/}
{/*</div>*/}

{/*<div style="display: none">*/}
{/*<img data-action="zoom" data-width="640" data-height="640" src="assets/img/instagram_13.jpg"/>*/}
{/*</div>*/}

{/*<div style="display: none">*/}
{/*<img data-action="zoom" data-width="1048" data-height="700" src="assets/img/unsplash_2.jpg"/>*/}
{/*</div>*/}
{/*</div>*/}

{/*<ul className="media-list mb-2">*/}
{/*<li className="media mb-3">*/}
{/*<img*/}
{/*className="media-object d-flex align-self-start mr-3"*/}
{/*src="assets/img/avatar-fat.jpg"/>*/}
{/*<div className="media-body">*/}
{/*<strong>Jacon Thornton: </strong>*/}
{/*Donec id elit non mi porta gravida at eget metus. Vivamus sagittis lacus vel augue laoreet*/}
{/*rutrum faucibus dolor auctor. Donec ullamcorper nulla non metus auctor fringilla. Praesent*/}
{/*commodo cursus magna, vel scelerisque nisl consectetur et. Sed posuere consectetur est at*/}
{/*lobortis.*/}
{/*</div>*/}
{/*</li>*/}
{/*<li className="media">*/}
{/*<img*/}
{/*className="media-object d-flex align-self-start mr-3"*/}
{/*src="assets/img/avatar-mdo.png"/>*/}
{/*<div className="media-body">*/}
{/*<strong>Mark Otto: </strong>*/}
{/*Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce dapibus, tellus ac cursus*/}
{/*commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus.*/}
{/*</div>*/}
{/*</li>*/}
{/*</ul>*/}
{/*</div>*/}
{/*</li>*/}

{/*<li className="media list-group-item p-4">*/}
{/*<img*/}
{/*className="media-object d-flex align-self-start mr-3"*/}
{/*src="assets/img/avatar-fat.jpg"/>*/}
{/*<div className="media-body">*/}
{/*<div className="media-body-text">*/}
{/*<div className="media-heading">*/}
{/*<small className="float-right text-muted">12 min</small>*/}
{/*<h6>Jacob Thornton</h6>*/}
{/*</div>*/}
{/*<p>*/}
{/*Donec id elit non mi porta gravida at eget metus. Integer posuere erat a ante venenatis dapibus*/}
{/*posuere velit aliquet. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur*/}
{/*ridiculus mus. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Lorem ipsum dolor*/}
{/*sit amet, consectetur adipiscing elit.*/}
{/*</p>*/}
{/*</div>*/}
{/*</div>*/}
{/*</li>*/}

{/*<li className="media list-group-item p-4">*/}
{/*<img*/}
{/*className="media-object d-flex align-self-start mr-3"*/}
{/*src="assets/img/avatar-mdo.png"/>*/}
{/*<div className="media-body">*/}
{/*<div className="media-heading">*/}
{/*<small className="float-right text-muted">34 min</small>*/}
{/*<h6>Mark Otto</h6>*/}
{/*</div>*/}

{/*<p>*/}
{/*Donec ullamcorper nulla non metus auctor fringilla. Vestibulum id ligula porta felis euismod*/}
{/*semper. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Etiam porta*/}
{/*sem malesuada magna mollis euismod. Donec sed odio dui.*/}
{/*</p>*/}

{/*<div className="media-body-inline-grid" data-grid="images">*/}
{/*<img style="display: none" data-width="640" data-height="640" data-action="zoom"*/}
{/*src="assets/img/instagram_3.jpg"/>*/}
{/*</div>*/}

{/*<ul className="media-list">*/}
{/*<li className="media">*/}
{/*<img*/}
{/*className="media-object d-flex align-self-start mr-3"*/}
{/*src="assets/img/avatar-dhg.png"/>*/}
{/*<div className="media-body">*/}
{/*<strong>Dave Gamache: </strong>*/}
{/*Donec id elit non mi porta gravida at eget metus. Vivamus sagittis lacus vel augue laoreet*/}
{/*rutrum faucibus dolor auctor. Donec ullamcorper nulla non metus auctor fringilla. Praesent*/}
{/*commodo cursus magna, vel scelerisque nisl consectetur et. Sed posuere consectetur est at*/}
{/*lobortis.*/}
{/*</div>*/}
{/*</li>*/}
{/*</ul>*/}
{/*</div>*/}
{/*</li>*/}
{/*</ul>*/}
{/*</div>*/}
{/*<div className="col-lg-3">*/}
{/*<div className="alert alert-warning alert-dismissible hidden-md-down" role="alert">*/}
{/*<button type="button" className="close" data-dismiss="alert" aria-label="Close"><span*/}
{/*aria-hidden="true">&times;</span></button>*/}
{/*<a className="alert-link" href="profile/index.html">Visit your profile!</a> Check your self, you aren't*/}
{/*looking well.*/}
{/*</div>*/}

{/*<div className="card mb-4 hidden-md-down">*/}
{/*<div className="card-block">*/}
{/*<h6 className="mb-3">Sponsored</h6>*/}
{/*<div data-grid="images" data-target-height="150">*/}
{/*<img className="media-object" data-width="640" data-height="640" data-action="zoom"*/}
{/*src="assets/img/instagram_2.jpg"/>*/}
{/*</div>*/}
{/*<p><strong>It might be time to visit Iceland.</strong> Iceland is so chill, and everything looks cool*/}
{/*here. Also, we heard the people are pretty nice. What are you waiting for?</p>*/}
{/*<button className="btn btn-outline-primary btn-sm">Buy a ticket</button>*/}
{/*</div>*/}
{/*</div>*/}

{/*<div className="card mb-4 hidden-md-down">*/}
{/*<div className="card-block">*/}
{/*<h6 className="mb-3">Likes*/}
{/*<small>· <a href="#">View All</a></small>*/}
{/*</h6>*/}
{/*<ul className="media-list media-list-stream">*/}
{/*<li className="media mb-2">*/}
{/*<img*/}
{/*className="media-object d-flex align-self-start mr-3"*/}
{/*src="assets/img/avatar-fat.jpg"/>*/}
{/*<div className="media-body">*/}
{/*<strong>Jacob Thornton</strong> @fat*/}
{/*<div className="media-body-actions">*/}
{/*<button className="btn btn-outline-primary btn-sm">*/}
{/*<span className="icon icon-add-user" /> Follow*/}
{/*</button>*/}
{/*</div>*/}
{/*</div>*/}
{/*</li>*/}
{/*<li className="media">*/}
{/*<a className="media-left" href="#">*/}
{/*<img*/}
{/*className="media-object d-flex align-self-start mr-3"*/}
{/*src="assets/img/avatar-mdo.png"/>*/}
{/*</a>*/}
{/*<div className="media-body">*/}
{/*<strong>Mark Otto</strong> @mdo*/}
{/*<div className="media-body-actions">*/}
{/*<button className="btn btn-outline-primary btn-sm">*/}
{/*<span className="icon icon-add-user"/> Follow*/}
{/*</button>*/}
{/*/!*</button>*!/*/}
{/*</div>*/}
{/*</div>*/}
{/*</li>*/}
{/*</ul>*/}
{/*</div>*/}
{/*<div className="card-footer">*/}
{/*Dave really likes these nerds, no one knows why though.*/}
{/*</div>*/}
{/*</div>*/}

{/*<div className="card card-link-list">*/}
{/*<div className="card-block">*/}
{/*© 2015 Bootstrap*/}
{/*<a href="#">About</a>*/}
{/*<a href="#">Help</a>*/}
{/*<a href="#">Terms</a>*/}
{/*<a href="#">Privacy</a>*/}
{/*<a href="#">Cookies</a>*/}
{/*<a href="#">Ads </a>*/}
{/*<a href="#">Info</a>*/}
{/*<a href="#">Brand</a>*/}
{/*<a href="#">Blog</a>*/}
{/*<a href="#">Status</a>*/}
{/*<a href="#">Apps</a>*/}
{/*<a href="#">Jobs</a>*/}
{/*<a href="#">Advertise</a>*/}
{/*</div>*/}
{/*</div>*/}
{/*</div>*/}
{/*</div>*/}
{/*</div>*/}
{/*</div>*/}
{/*);*/}
{/*};*/}