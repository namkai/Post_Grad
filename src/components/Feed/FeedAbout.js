import React from 'react';
import FeedBody from './FeedBody';
import FeedHeader from './FeedHeader';
import FeedPhotos from './FeedPhotos';

const ProfileAbout = () => (
	<div className="col-lg-3">
		<FeedHeader />
		<FeedBody />
		<FeedPhotos />
	</div>
);

export default ProfileAbout;
