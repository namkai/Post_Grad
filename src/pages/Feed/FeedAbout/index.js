import React from 'react';
import { connect } from 'react-redux';

import FeedBody from './FeedBody';
import FeedHeader from './FeedHeader';
import FeedPhotos from './FeedPhotos';
import ListModal from './ListModal';

const ProfileAbout = ({ user, friends, projects }) => (
	<div className="col-lg-3">
		<ListModal title={ 'Friends' } list={ friends } id={ 'userModal' } />
		<ListModal title={ 'Projects' } list={ projects.projects } id={ 'projectsModal' } />
		<FeedHeader user={ user } friends={ friends.length } projects={ projects.projects.length } />
		<FeedBody />
		<FeedPhotos />
	</div>
);

const mapStateToProps = ({ user, friends, projects }) => ({
	user,
	friends,
	projects,
});

export default connect(mapStateToProps)(ProfileAbout);
