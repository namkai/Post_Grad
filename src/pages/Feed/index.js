import React from 'react';
import FeedAbout from './FeedAbout/';
import FeedStream from './FeedStream'

export const Profile = () => (
	<div>
		<div className="container pt-4">
			<div className="row">
				<FeedAbout />
				<FeedStream />
			</div>
		</div>
	</div>
);

export default Profile;
