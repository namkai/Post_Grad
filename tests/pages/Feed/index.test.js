import React from 'react';
import { Profile } from '../../../src/Pages/Feed';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Profile', () => {
	it('Renders a profile', () => {
		const profile = shallow(<Profile />);
		expect(toJson(profile)).toMatchSnapshot();
	});
});