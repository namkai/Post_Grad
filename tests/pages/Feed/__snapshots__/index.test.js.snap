exports[`Profile Renders a profile 1`] = `
<div>
  <div
    className="container pt-4">
    <div
      className="row">
      <Connect(ProfileAbout) />
      <Connect(FeedStream) />
    </div>
  </div>
</div>
`;
